grblHAL/STM32F4xx
# PlatformIO: initial support for building firmware with pio

This provides four target environments:

- `nucleo_f411re_morpho_cnc` (Nucleo STM32F411 with Morpho CNC shield)
- `nucleo_f411re_protoneer` (dito buto with Protoneer CNC shield for Arduino)
- `nucleo_f446re_morpho_cnc`
- `nucleo_f446re_protoneer`

To test this out, install PlatformIO core:

- https://docs.platformio.org/en/latest/core/installation.html

Example:

    # Build all target environments
    pio run
    # Build a single target environments
    pio run -e nucleo_f411re_protoneer
    # Upload the firmware (or alternatively, copy .pio/build/<env name>/firmware.bin to the USB MSD)
    pio run -e nucleo_f411re_protoneer -t upload

I've only tested `pio run -e nucleo_f411re_protoneer` and but believe the other targets should work as well.  I haven't included all of the targets from the `.launch` files since I lack the necessary hardware to test them out.  I haven't synced the build flags that may or may not be present in the `.cproject` file created by the original IDE.

New boards can be added by duplicating an existing environment and adjusting:

- `board` (find an existing one on https://docs.platformio.org/en/latest/platforms/ststm32.html#boards and use the value from the board-specific page)
- `board_build.ldscript`
- `build_flags`
- `lib_deps`

In the future, automated build tests may be implemented using GitHub Actions --  see https://docs.platformio.org/en/latest/integration/ci/github-actions.html
